clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerMembershipManager.java

```diff
@@ -469,8 +493,17 @@ public int joinGroupEpoch() {
      */
     @Override
     public int leaveGroupEpoch() {
-        return groupInstanceId.isPresent() ?
-                ConsumerGroupHeartbeatRequest.LEAVE_GROUP_STATIC_MEMBER_EPOCH :
-                ConsumerGroupHeartbeatRequest.LEAVE_GROUP_MEMBER_EPOCH;
+        boolean isStaticMember = groupInstanceId.isPresent();
+        if (REMAIN_IN_GROUP == leaveGroupOperation && isStaticMember) {
```

Comment:
this condition is basically redundant, right? if `isStaticMember` is true then we'll return `LEAVE_GROUP_STATIC_MEMBER_EPOCH` anyways via the final check